### PR TITLE
RapidSmith 2 TINCR checkpoints 

### PR DESCRIPTION
**Pull request by ttown523**
I have updated the write_checkpoints.tcl file to have a "write_rscp" function. This will create a checkpoint in the format that RapidSmith2 is expecting. All other write_tcp functions were restored to their original state before any RS2 modifications were made.

### CODE REVIEW

**Comment by ttown523** [comment]
I do have one question. In my new functions that I created, the function call `[get_name $cell]` no longer works. This seems to be a scoping issue, but I am not familiar enough with TCL to understand how to fix it. For now, I changed these calls to `[tincr::cells::get_name]` (or something similar). Any suggestions on how to fix this issue?

File: tincr/io/design/tincr_checkpoints.tcl [hunk 1]
```diff
@@ -13,12 +13,14 @@ namespace eval ::tincr:: {
         write_design_info \
         write_placement_xdc \
         write_routing_xdc \
-        get_design_info
+        get_design_info \
+		write_rscp
```
Comment (bradselw) [inline tincr/io/design/tincr_checkpoints.tcl]: Prefer spaces instead of tabs

**Comment by bradselw** [comment]
This looks good to me. Just make sure you go through and replace all the tabs with 4 spaces and make the suggested changes. Just FYI, you don't have to create another pull request, just add your changes to this one by creating a new commit and pushing it to your master branch on GitHub.

**Comment by bradselw** [comment]
I'll look into the get_name issue this evening.

File: tincr/io/design/tincr_checkpoints.tcl [hunk 2]
```diff
@@ -208,3 +205,91 @@ proc ::tincr::get_design_info {args} {
...
+				#TODO These get_*_info commands should be deprecated
+				lappend pins_to_lock "[::tincr::pins::info $pin name]:[::tincr::bel_pins::get_info $bel_pin name]"
+			}
+		} 
+	}
+	return $pins_to_lock
+}
+
+
+proc write_placement_rs2 {filename} {
```
Comment (bradselw) [inline tincr/io/design/tincr_checkpoints.tcl]: `proc ::tincr::write_placement_rs2 {filename} {`

File: tincr/io/design/tincr_checkpoints.tcl [hunk 3]
```diff
@@ -208,3 +205,91 @@ proc ::tincr::get_design_info {args} {
     
     return ""
 }
+
+proc get_pins_to_lock {cell} {
```
Comment (bradselw) [inline tincr/io/design/tincr_checkpoints.tcl]: `proc ::tincr::get_pins_to_lock {cell} {`

File: tincr/io/design/tincr_checkpoints.tcl [hunk 4]
```diff
@@ -208,3 +205,91 @@ proc ::tincr::get_design_info {args} {
...
+				puts $txt "LOCK_PINS $pins_to_lock [get_cells [get_property NAME $cell]]"
+        	}
+        }
+    }
+	
+    close $txt
+}
+
+#TODO: Update this once we have a better idea of what we will need for RS2 in terms of routing.
+proc write_routing_rs2 {args} {
```
Comment (bradselw) [inline tincr/io/design/tincr_checkpoints.tcl]: `proc ::tincr::write_routing_rs2 {args} {`

File: tincr/io/design/tincr_checkpoints.tcl (hunk 1 above)
Comment (bradselw) [inline tincr/io/design/tincr_checkpoints.tcl]: Add `get_pins_to_lock`, `write_placement_rs2`, and `write_routing_rs2`

**Comment by bradselw** [comment]
The reason `get_name` wasn't working in your functions is because they were being declared in the global namespace (which is generally a bad idea), and `get_name` is declared in the `tincr` namespace. Make sure you prefix each of your procs' names with `::tincr::` and also add them to the `namespace export` list.

**Comment by ttown523** [comment]
Ok, I made the changes you suggested. Let me know if I need to change anything else.

**Comment by bradselw** [comment]
Thanks, merged.

**Pull request merged**